drivers/metal/metal_objects.mm

```diff
@@ -212,24 +212,23 @@
 void MDCommandBuffer::render_bind_uniform_sets(VectorView<RDD::UniformSetID> p_uniform_sets, RDD::ShaderID p_shader, uint32_t p_first_set_index, uint32_t p_set_count, uint32_t p_dynamic_offsets) {
 	DEV_ASSERT(type == MDCommandBufferStateType::Render);
 
-	if (render.dynamic_offsets_mask != p_dynamic_offsets) {
-		render.dirty.set_flag(RenderState::DIRTY_UNIFORMS);
-		render.uniform_set_mask |= UINT64_MAX; // We don't know which ones are dirty. Potentially all of them.
-		render.dynamic_offsets_mask = p_dynamic_offsets;
+	render.dynamic_offsets |= p_dynamic_offsets;
```

Comment:
Multiple calls to `render_bind_uniform_sets` may occur before the draw call, so we need merge the value of `p_dynamic_offsets`. Each nibble refers to the `frame_index` captured at bind time, which you added in your recent commit.